pkgs/development/embedded/openocd-rp2040/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, fetchgit
+, pkg-config
+, hidapi
+, libftdi1
+, libusb1
+, which
+, libtool
+, autoconf
+, automake
+, texinfo
+, git
+, libgpiod
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openocd-rp2040";
+  version = "0.11.0";
+  src = fetchgit {
+    url = "https://github.com/raspberrypi/openocd";
+    rev = "228ede43db3665e470d2e518730de013a8c74411";
+    sha256 = "sha256-NrN01BdkLlM9nHilpOSarJvlDWFCVMVInLP/l7GoCIc=";
+    leaveDotGit = true;
```

Comment:
why are you leaving the dot git folder?

Reply:
It's been quite a while, since I've written this derivation, but I'm pretty sure that the call to `./bootstrap`  does some git stuff, so it relies on the dotgit folder still being there.

Reply:
We usually want to avoid using `leaveDotGit` as much as possible to the point where we write patches to patch out sections that would need it since `leaveDotGit` isn't really reproducible.

Reply:
I'm unable to remove `leaveDotGit`.
```
Setting up submodules
fatal: not a git repository (or any of the parent directories): .git
error: builder for '/nix/store/jvz4qr7nad375lm3iq5fpgrck0i826vg-openocd-rp2040-0.12.0.drv' failed with exit code 128;
       last 10 log lines:
       > configure.ac:42: installing 'build-aux/config.guess'
       > configure.ac:42: installing 'build-aux/config.sub'
       > configure.ac:19: installing 'build-aux/install-sh'
       > configure.ac:19: installing 'build-aux/missing'
       > Makefile.am: installing './INSTALL'
       > Makefile.am: installing 'build-aux/depcomp'
       > Makefile.am:26: installing 'build-aux/mdate-sh'
       > Makefile.am:26: installing 'build-aux/texinfo.tex'
       > Setting up submodules
       > fatal: not a git repository (or any of the parent directories): .git
```

Reply:
Removed `leaveDotGit`.